avocado/plugins/list.py

```diff
@@ -133,6 +129,39 @@ def _display_extra(suite, verbose=True):
             for key in sorted(suite.tags_stats):
                 LOG_UI.info("%s: %s", key, suite.tags_stats[key])
 
+    @staticmethod
+    def _matches_tag_filter(runnable, filter_by_tags):
```

Comment:
The filtering itself is a bit more complex, but there's a function you can reuse in `avocado/core/tags.py`.  You could use it like:

```diff
diff --git a/avocado/plugins/list.py b/avocado/plugins/list.py
index 2cf67e26d..156c2ed48 100644
--- a/avocado/plugins/list.py
+++ b/avocado/plugins/list.py
@@ -22,6 +22,7 @@ from avocado.core.plugin_interfaces import CLICmd
 from avocado.core.resolver import ReferenceResolutionResult
 from avocado.core.settings import settings
 from avocado.core.suite import TestSuite
+from avocado.core.tags import filter_tags_on_runnables
 from avocado.utils.astring import iter_tabular_output
 
 
@@ -142,8 +143,18 @@ class List(CLICmd):
         """
         test_matrix = []
         verbose = suite.config.get("core.verbose")
+        tags = suite.config.get("filter.by_tags.tags")
+        include_empty = suite.config.get("filter.by_tags.include_empty")
+        include_empty_key = suite.config.get("filter.by_tags.include_empty_key")
+
         for resolution in suite.resolutions:
-            for runnable in resolution.resolutions:
+            if tags:
+                runnables = filter_tags_on_runnables(
+                    [resolution], tags, include_empty, include_empty_key
+                )
+            else:
+                runnables = resolution.resolutions
+            for runnable in runnables:
                 if verbose:
                     tags = runnable.tags or {}
                     test_matrix.append(
```